### PR TITLE
Yolo/rework addr

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- rename `to_http_domain_url` to `to_http_url_with_path` to better reflect what the method do
- rename `BaseBackendAddr::to_http_redirection_url` to `to_http_url` to better reflect what the method do
- correct some typo
- don't publicly expose `_to_url`
- more consistent spacing

### CODE REVIEW

**Comment by touilleMan** [comment]
> rename BaseBackendAddr::to_http_redirection_url to to_http_url to better reflect what the method do

I disagree, redirection url is a special case and we don't want to mess it with e.g. the http url used to send api command to the server.

**Comment by FirelightFlagboy** [comment]
This method do nothing related to redirection

File: oxidation/crates/parsec_api_types/src/addr.rs [hunk 1]
```diff
@@ -211,24 +212,22 @@ impl BaseBackendAddr {
         url
     }
 
-    pub fn to_http_redirection_url(&self) -> Url {
+    /// Create a url for http request.
+    pub fn to_http_url(&self) -> Url {
```
Comment (touilleMan) [inline oxidation/crates/parsec_api_types/src/addr.rs:216]: I think we said we could remove entirely this and only use `BaseBackendAddr::to_http_url_with_path(None)` instead

Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_types/src/addr.rs:216]: I was asking myself, if I wouldn't change its (for `to_http_url_with_path`) signature to

```rust
pub fn to_http_url_with_path(&self, path: &str) -> Url {}
```

Comment (touilleMan) [reply oxidation/crates/parsec_api_types/src/addr.rs:216]: > my comment above was for the case when we have both to_http_url and to_http_url_with_path

I think it's better to only have a single `to_http_url_with_path(path: Option<...>)`

Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_types/src/addr.rs]: Done, I've merged both

**Review by touilleMan — APPROVED** [review]

**Pull request merged**